ansible/bootstrap.yml

```diff
@@ -227,24 +227,24 @@
     - update
   tasks:
     - name: Check for pending reboot from package updates
-      stat: 
-        path: /var/run/reboot-required
+      command:
+        cmd: dnf needs-restarting
```

Comment:
Do we need -r here E.g:
```
[rocky@cclr-dev-control ~]$ sudo dnf needs-restarting -r
No core libraries or services have been updated since boot-up.
Reboot should not be necessary.
```
No idea what it outputs when you do need a reboot mind :D

Reply:
OK, man reveals all:
```
DNF-NEEDS-RESTARTING(8)                                                                        dnf-plugins-core                                                                        DNF-NEEDS-RESTARTING(8)

NAME
       dnf-needs-restarting - DNF needs_restarting Plugin

       Check for running processes that should be restarted.

SYNOPSIS
       dnf needs-restarting [-u] [-r]

DESCRIPTION
       needs-restarting  looks  through  running  processes and tries to detect those that use files from packages that have been updated after the given process started. Such processes are reported by this
       tool.

       Note that in most cases a process should survive update of its binary and libraries it is using without requiring to be restarted for proper operation. There are however specific cases when this does
       not apply. Separately, processes often need to be restarted to reflect security updates.

OPTIONS
       All general DNF options are accepted, see Options in dnf(8) for details.

       -u, --useronly
          Only consider processes belonging to the running user.

       -r, --reboothint
          Only report whether a reboot is required (exit code 1) or not (exit code 0).
```

Reply:
Just looks like mine has some false positives in stdout:
```
[rocky@cclr-dev-control ~]$ sudo dnf needs-restarting
785 : /sbin/auditd 
1054 : /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only 
1152 : /usr/lib/polkit-1/polkitd --no-debug 
1367 : /usr/sbin/rpc.statd 
6476 : (sd-pam) 
6559 : /usr/bin/dbus-daemon --session --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only
``` 

or maybe they did get updated :thinking:

Reply:
But that conflicts with:
```
[rocky@cclr-dev-control ~]$ sudo dnf needs-restarting
785 : /sbin/auditd 
1054 : /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only 
1152 : /usr/lib/polkit-1/polkitd --no-debug 
1367 : /usr/sbin/rpc.statd 
6476 : (sd-pam) 
6559 : /usr/bin/dbus-daemon --session --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only 
[rocky@cclr-dev-control ~]$ sudo dnf needs-restarting -r
No core libraries or services have been updated since boot-up.
Reboot should not be necessary.
```

Reply:
I'd be tempted to trust the reboot hint - what do you reckon?